Help flag should show task help when task is specified (#30275)
gradle/gradle: platforms/documentation/docs/src/docs/release/notes.md

The `cpp-unit` and `xctest` plugins are not yet compatible.

### Other improvements

> #### Help task

Reviewer: ```suggestion
#### Updates to task-specific help on the CLI
```

> In any version of Gradle, you can run `gradle help --task <task-name>` to get help on a specific task.

Reviewer: ```suggestion
Previously, you could run `gradle help --task <task-name>` to get help on a specific task.
```

> You can now get the same result with `gradle <task-name> --help`, e.g. `gradle init --help`.

Reviewer: ```suggestion
Now, you can achieve the same result more conveniently with `gradle <task-name> --help`.
For example, use `gradle init --help` to get information about the `init` task.
```

> This is slightly more convenient and consistent with other command-line tools.

Reviewer: ```suggestion

This approach is not only more concise but also aligns with the conventions of other command-line tools.
```
